### PR TITLE
Update modulemgr

### PR DESCRIPTION
**Pull request by Princess-of-Sleeping**

### CODE REVIEW

**Comment by d3m3vilurr** [comment]
although this patch can breaking the kernel modules via changing the names, LGTM this patch and it seems to use more suitable names.

File: include/psp2/kernel/modulemgr.h [hunk 1]
```diff
@@ -28,27 +28,28 @@ extern "C" {
...
-	void *vaddr;    //!< address in memory
-	SceUInt memsz;  //!< size in memory
-	SceUInt flags;  //!< meaning unknown
-	SceUInt res;    //!< unused?
+  SceSize size;   //!< this structure size (0x18)
+  SceUInt perms;  //!< probably rwx in low bits
+  void *vaddr;    //!< address in memory
+  SceSize memsz;  //!< size in memory
+  SceSize filesz; //!< original size of memsz
+  SceUInt res;    //!< unused
```
Comment (d3m3vilurr) [inline include/psp2/kernel/modulemgr.h]: could you keep the `tab` char in here?

File: include/psp2kern/kernel/modulemgr.h [hunk 2]
```diff
@@ -27,38 +27,39 @@ extern "C" {
...
+typedef struct SceKernelSegmentInfo {
+  SceSize size;   //!< this structure size (0x18)
   SceUInt perms;  //!< probably rwx in low bits
   void *vaddr;    //!< address in memory
-  SceUInt memsz;  //!< size in memory
-  SceUInt flags;  //!< meaning unknown
-  SceUInt res;    //!< unused?
+  SceSize memsz;  //!< size in memory
+  SceSize filesz; //!< original size of memsz
+  SceUInt res;    //!< unused
```
Comment (d3m3vilurr) [inline include/psp2kern/kernel/modulemgr.h]: same here. please don't fix hardtab & softtab.

File: include/psp2/kernel/modulemgr.h [hunk 3]
```diff
@@ -28,27 +28,28 @@ extern "C" {
...
-	void *module_exit;
+	void *start_entry;
+	void *stop_entry;
+	void *exit_entry;
 	void *exidxTop;
 	void *exidxBtm;
-	SceUInt unk40;
-	SceUInt unk44;
+	void *extabTop;
+	void *extabBtm;
```
Comment (d3m3vilurr) [inline include/psp2/kernel/modulemgr.h]: sorry, last thing.
we already break the backward compatibility.
could you change these `camelCase` to `snake_case` ?

**Comment by d3m3vilurr** [comment]
@Princess-of-Sleeping sorry, my late & delay reviews.
thank you your contributions :+1:

**Comment by yne** [comment]
@d3m3vilurr can you explain why we switched the SceKernelModuleName from a `char[]` to a `struct{char[]}` ?

**Pull request merged**